fe/fe-core/src/main/java/org/apache/doris/datasource/lakesoul/LakeSoulUtils.java

```diff
@@ -0,0 +1,352 @@
+    public static final String CDC_COLUMN = "lakesoul_cdc_change_column";
+    public static final String LIST_DELIM = ";";
+    public static final String PARTITIONS_KV_DELIM = "=";
+    public static final String FS_S3A_ACCESS_KEY = "fs.s3a.access.key";
+    public static final String FS_S3A_SECRET_KEY = "fs.s3a.secret.key";
+    public static final String FS_S3A_ENDPOINT = "fs.s3a.endpoint";
+    public static final String FS_S3A_REGION = "fs.s3a.endpoint.region";
+    public static final String FS_S3A_PATH_STYLE_ACCESS = "fs.s3a.path.style.access";
+
+    public static List<PartitionInfo> applyPartitionFilters(
+            List<PartitionInfo> allPartitionInfo,
+            String tableName,
+            Schema partitionArrowSchema,
+            Map<String, ColumnRange> columnNameToRange
+    ) throws IOException {
+
+        Expression conjunctionFilter = null;
+        for (Field field : partitionArrowSchema.getFields()) {
+            ColumnRange columnRange = columnNameToRange.get(field.getName());
+            if (columnRange != null) {
+                Expression expr = columnRangeToSubstraitFilter(field, columnRange);
+                if (expr != null) {
+                    if (conjunctionFilter == null) {
+                        conjunctionFilter = expr;
+                    } else {
+                        conjunctionFilter = SubstraitUtil.and(conjunctionFilter, expr);
+                    }
+                }
+            }
+        }
+        return SubstraitUtil.applyPartitionFilters(
+            allPartitionInfo,
+            partitionArrowSchema,
+            SubstraitUtil.substraitExprToProto(conjunctionFilter, tableName)
+        );
+    }
+
+    public static Expression columnRangeToSubstraitFilter(
+            Field columnField,
+            ColumnRange columnRange
+    ) throws IOException {
+        Optional<RangeSet<ColumnBound>> rangeSetOpt = columnRange.getRangeSet();
+        if (columnRange.hasConjunctiveIsNull() || !rangeSetOpt.isPresent()) {
+            return SubstraitUtil.CONST_TRUE;
+        } else {
+            RangeSet<ColumnBound> rangeSet = rangeSetOpt.get();
+            if (rangeSet.isEmpty()) {
+                return SubstraitUtil.CONST_TRUE;
+            } else {
+                Expression conjunctionFilter = null;
+                for (Range range : rangeSet.asRanges()) {
+                    Expression expr = rangeToSubstraitFilter(columnField, range);
+                    if (expr != null) {
+                        if (conjunctionFilter == null) {
+                            conjunctionFilter = expr;
+                        } else {
+                            conjunctionFilter = SubstraitUtil.or(conjunctionFilter, expr);
+                        }
+                    }
+                }
+                return conjunctionFilter;
+            }
+        }
+    }
+
+    public static Expression rangeToSubstraitFilter(Field columnField, Range range) throws IOException {
```

Comment:
I suggest to write unit test for these expr conversion logic.
You can refer to: IcebergPredicateTest.java